core/dbt/parser/unit_tests.py

```diff
@@ -438,6 +437,31 @@ def process_models_for_unit_test(
     target_model_id = unit_test_def.depends_on.nodes[0]
     target_model = manifest.nodes[target_model_id]
     assert isinstance(target_model, ModelNode)
+
+    target_model_is_incremental = "macro.dbt.is_incremental" in target_model.depends_on.macros
+    unit_test_def_has_incremental_override = unit_test_def.overrides and isinstance(
+        unit_test_def.overrides.macros.get("is_incremental"), bool
+    )
+
+    if target_model_is_incremental and (not unit_test_def_has_incremental_override):
+        raise ParsingError(
+            f"Boolean override for 'is_incremental' must be provided for unit testing model '{target_model.name}'"
```

Comment:
suggestion -- let's also add the unit test name here for ease of user debugging since there can be multiple unit tests for a single model